parsing/lexer.mll

```diff
@@ -141,6 +141,16 @@ let print_warnings = ref true
 let store_escaped_char lexbuf c =
   if in_comment () then store_lexeme lexbuf else store_string_char c
 
+let store_escaped_uchar =
+  let uchar_utf_8_enc = Buffer.create 4 in
+  fun lexbuf u ->
+    if in_comment () then store_lexeme lexbuf else
+    begin
+      Buffer.reset uchar_utf_8_enc;
+      Buffer.add_utf_8_uchar uchar_utf_8_enc u;
+      store_string (Buffer.contents uchar_utf_8_enc);
```

Comment:
Not directly related to this PR, but does anyone know why lexer.mml doesn't simply use a Buffer.t instead of its own version?  This would simplify the chunk above.